src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp

```diff
@@ -508,11 +493,21 @@ bool SnippetsMarkSkipped::run_on_model(const std::shared_ptr<ov::Model> &m) {
             }
             SetNodeFusingType(node, NodeFusingType::FusedWithMisc);
         } else if (isSuitableMatMulParent(node)) {
-            if (canBeMatMulExecutedInInt8(node->get_input_element_type(0), node->get_input_element_type(1)))
-                SetNodeFusingType(node, NodeFusingType::FusedWithMatMulI8);
-            else
-                SetNodeFusingType(node, NodeFusingType::FusedWithMatMul);
-            channelAxis = DEFAULT_AXIS;
+            const auto out_acts = node->input_value(0);
+            const auto out_weights = node->input_value(1);
+            const auto rank_a = out_acts.get_partial_shape().rank().get_length();
+            const auto rank_w = out_weights.get_partial_shape().rank().get_length();
+            const auto out_rank = node->get_output_partial_shape(0);
```

Comment:
Should be `const auto out_rank = node->get_output_partial_shape(0).rank();` maybe?